main/samba/samba.initd

```diff
@@ -1,24 +1,41 @@
 #!/sbin/openrc-run
 
 extra_started_commands="reload"
+piddir=${piddir:-"/run/samba"}
+daemon_list="${daemon_list//samba/}"
 
 DAEMON=${SVCNAME#samba.}
 if [ "$DAEMON" != "samba" ]; then
 	daemon_list=$DAEMON
+elif [ "${daemon_list// /}" == "" ] || [ -z '$daemon_list' ]; then
```

Comment:
The `== ` is bashism. use single `=`.

The `[ -z '$daeom_list' ]`  will always return `false`. The single quotes will prevent `$daemon_list` to get expanded and return the literal string `$daemon_list`. Its equivalent to `"\$daemon_list"`. This is most likely not what you intend.

Reply:
I want make sure that samba launches only when $daemon_list is empty (with or without spaces) and not set. I'm aware of the situation with single quotes  but it seems that it misled me while the expression also returns true because '$daemon_list' didn't set as well. If $daemon_list set by any string, /bin/sh seems to append $daemon_list's value to "-z" rather than treating it as a variable identifier. I'm open for any simple solution.